### PR TITLE
Docker prune automation

### PR DESCRIPTION
**Pull request by giorgiosironi**
/cc @diversemix try it out?

### CODE REVIEW

File: elife/docker-scripts/docker-prune [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
-#docker images prune
 echo "Clean up all images not used in the last 14 days"
+limit=$(date +%Y-%m-%d -d "14 days ago")
+docker image prune -a --filter "until=$limit" -f
+
 # https://github.com/moby/moby/issues/9054#issuecomment-227920187
-docker images | grep -v REPOSITORY | awk '{print $3}' | xargs -n 1 docker history | grep -v missing | grep -v "^IMAGE" | egrep -v "day|hour|minute|second" | awk '{print $1}' | sort | uniq | xargs -n 1 docker inspect | jq -r .[0].RepoTags[] | xargs docker rmi
+#images="/tmp/$(date +%Y-%m-%dT%H:%M:%S).txt"
+#docker images | grep -v REPOSITORY | awk '{print $3}' | xargs -n 1 docker history | grep -v missing | grep -v "^IMAGE" | egrep -v "day|hour|minute|second" | awk '{print $1}' | sort | uniq > "$images"
+#cat "$images" | xargs -n 1 docker inspect | jq -r .[0].RepoTags[] | xargs docker rmi
```
Comment (thewilkybarkid) [inline elife/docker-scripts/docker-prune]: Can remove all these comments now?

File: elife/docker-scripts/docker-prune [hunk 2]
```diff
@@ -1,10 +1,15 @@
 #!/bin/bash
 set -e
 
-docker container prune
-docker volume prune
+docker container prune -f
+docker network prune -f
+docker volume prune -f
```
Comment (thewilkybarkid) [inline elife/docker-scripts/docker-prune]: Clearer to use `--force`.

File: elife/docker-scripts/docker-prune [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
 
-#docker images prune
 echo "Clean up all images not used in the last 14 days"
+limit=$(date +%Y-%m-%d -d "14 days ago")
```
Comment (thewilkybarkid) [inline elife/docker-scripts/docker-prune]: Unrelated, but macOS uses BSD's `date` so has to be changed to `date -j -v -14d "+%Y-%m-%d"` to reuse.

File: elife/docker-scripts/docker-prune (hunk 1 above)
Comment (giorgiosironi) [reply elife/docker-scripts/docker-prune]: They are in a commit now (some new code I tried out) so yes

File: elife/docker-scripts/docker-prune (hunk 3 above)
Comment (giorgiosironi) [reply elife/docker-scripts/docker-prune]: Is there a portable option to generate `2018-07-27`? This is usually deployed on Ubuntu Linux, other flavors may be used e.g. in EKS nodes that thave their own AMI.

Comment (thewilkybarkid) [reply elife/docker-scripts/docker-prune]: Could try https://stackoverflow.com/a/8748193/77199.

Comment (giorgiosironi) [reply elife/docker-scripts/docker-prune]: Seems `docker image prune` supports `"until=240h"` so no `date` needed

File: elife/docker-scripts/docker-prune [hunk 4]
```diff
@@ -1,10 +1,9 @@
...
-docker volume prune
+docker container prune --force
+docker network prune --force
+docker volume prune --force
 
-#docker images prune
 echo "Clean up all images not used in the last 14 days"
-# https://github.com/moby/moby/issues/9054#issuecomment-227920187
-docker images | grep -v REPOSITORY | awk '{print $3}' | xargs -n 1 docker history | grep -v missing | grep -v "^IMAGE" | egrep -v "day|hour|minute|second" | awk '{print $1}' | sort | uniq | xargs -n 1 docker inspect | jq -r .[0].RepoTags[] | xargs docker rmi
+docker image prune --all --filter "until=336h" --force
```
[the remark this answers is not in the document]
Comment (giorgiosironi) [reply elife/docker-scripts/docker-prune]: `Error response from daemon: strconv.ParseInt: parsing "14d": invalid syntax`

[the remark this answers is not in the document]
Comment (giorgiosironi) [reply elife/docker-scripts/docker-prune]: (because `336h` is an int?) :thinking:

[the remark this answers is not in the document]
Comment (giorgiosironi) [reply elife/docker-scripts/docker-prune]: Source https://docs.docker.com/engine/reference/commandline/image_prune/#filtering

**Pull request merged**